docs/build-on-polkadot/xcm/asset-transfer-api.md

```diff
@@ -0,0 +1,491 @@
+    ```
+
+2. Initialize a new project:
+    ```bash
+    npm init -y
+    ```
+
+Once you have a project set up, you can install the asset-transfer-api package:
+
+```bash
+npm install @substrate/asset-transfer-api@{{build_on_polkadot.xcm.asset_transfer_api.version}}
+```
+
+!!!note
+    This documentation covers version `{{build_on_polkadot.xcm.asset_transfer_api.version}}` of Asset Transfer API.
+
+## Set Up Asset Transfer API
+
+To initialize the Asset Transfer API, you need three key components:
+
+- A Polkadot.js API instance
+- The `specName` of the chain
+- The XCM version to use
+
+<!-- You can set up the API in two ways. Choose the option that best fits your project's structure and requirements. Both methods will result in a fully initialized AssetTransferApi instance ready for use:
+
+### Using an Existing Polkadot.js API
+
+If you already have an initialized Polkadot.js API instance, you can use it to set up the Asset Transfer API:
+
+```javascript
+// Assuming 'api' is your initialized Polkadot.js API instance
+const { specName } = await api.rpc.state.getRuntimeVersion();
+const safeXcmVersion = await fetchSafeXcmVersion(api);
+
+const assetsApi = new AssetTransferApi(api, specName, safeXcmVersion);
+```
+-->
+
+### Using helper function from library
+
+For a simpler setup process, you can leverage the `constructApiPromise` helper function provided by the library. 
+
+<!-- This method automatically initializes the Polkadot.js API, retrieves the specName, and determines the safe XCM version. -->
+
+```javascript
+import { AssetTransferApi, constructApiPromise } from '@substrate/asset-transfer-api';
+
+async function main() {
+    const { api, specName, safeXcmVersion } = await constructApiPromise('<INSERT_WEBSOCKET_URL>');
+
+    const assetsApi = new AssetTransferApi(api, specName, safeXcmVersion);
+
+    // Your code using assetsApi goes here
+}
+
+main();
+```
+
+!!!note
+    The code examples are enclosed in an async main function to provide the necessary asynchronous context. However, if you're already working within an async environment, you can use the code directly. The key is to ensure you're in an async context when working with these asynchronous operations, regardless of your specific setup.
+
+
+## Asset Transfer API Overview
+
+The AssetTransferApi provides a powerful method for cross-chain asset transfers: `createTransferTransaction`. This function allows you to create XCM transactions for transferring assets or native tokens between different chains.
+This method takes several parameters to specify the details of the transfer, including the destination chain, recipient address, assets to be transferred, their amounts and an optional parameter for further customization. It is able to infer what kind of transaction is necessary given the inputs. When sending cross-chain transfers, the API performs extensive validation to ensure the inputs are valid, and the assets either exist or don't.
```

Comment:
```suggestion
The AssetTransferApi provides a powerful method for cross-chain asset transfers: `createTransferTransaction`. This function allows you to create XCM transactions to transfer assets or native tokens between different chains. 
This method takes several parameters to specify the transfer details, including the destination chain, recipient address, assets to be transferred, and their amounts, as well as an optional parameter for further customization. It can infer what kind of transaction is necessary given the inputs. When sending cross-chain transfers, the API performs extensive validation to ensure the inputs are valid and the assets either exist or don't.
```